pkg/kubelet/cm/cpuset/cpuset.go

```diff
@@ -0,0 +1,260 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cpuset
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/golang/glog"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+// CPUSet is a set-like data structure for CPU IDs.
+type CPUSet map[int]struct{}
```

Comment:
I'm not totally convinced it's worth it to use an unsigned type here, in terms of the inconvenience to calling code. In practice, on any real system the valid elements are already a tiny subset of `int`, and `int` is similarly guaranteed to be at least 32 bits wide.